# **HYPOTHESIS TESTING**


 **1. State the Hypotheses statement:**

**Null Hypothesis Ho**: The mean weekly operating cost is W = $1000 + $5X

**Alternative Hypothesis Ha**: The mean weekly operating cost is higher than the theoretical model,  W > 1000 +5X
(right Tailed Test)

In [ ]:
import numpy as np
import scipy.stats as stats


**2. Calculate the Test Statistic:**

Given:
- Sample mean weekly cost  = 3050
- Theoretical mean weekly cost (mu) = ( 1000 + 5 times * 600 = 4000 )
- Standard deviation (sigma ) = ( 5 times * 25 = 125 )
- Sample size ( n ) = 25

The formula for the test statistic is:

t = (x̄ - μ) / (s / √n)


In [ ]:
z_cal=(3050-4000)/(125/np.sqrt(25))
z_cal

-38.0

**3. Determine the Critical Value:**

alpha level of 5% (α = 0.05)
,determine the critical value from the standard normal (Z) distribution table.


In [ ]:
z_tab = stats.norm.ppf(1-0.05)
z_tab

1.6448536269514722

In [ ]:
if z_cal < z_tab:
    print("Accept H0")
    print("The mean weekly operating cost is W = 1000+ 5X")
else:
  print("Accept H1")
  print("The mean weekly operating cost is higher than the theoretical model, W > 1000 +5X ")

Accept H0
The mean weekly operating cost is W = 1000+ 5X


**4. Decision:**

If z_cal > z_tab (t>1.645), reject the null hypothesis.

If z_cal < z_tab (t≤1.645), accept the null hypothesis.

**5. Conclusion:**
Given that the calculated test statistic is much smaller than the critical value, there is insufficient evidence to support the claim that the weekly operating costs are higher than the model suggests. Therefore, the null hypothesis 𝐻o is Accepted.

Accept Ho.

# **CHI-SQUARE TEST**





**1. State the Hypotheses:**


*   Null Hypothesis (𝐻0): There is no significant association between the type of smart home device purchased and customer satisfaction level.
In other words, the satisfaction levels are independent of the device type.
* Alternative Hypothesis (𝐻1​): There is a significant association between the type of smart home device purchased and customer satisfaction level.
In other words, the satisfaction levels are not independent of the device type.




In [ ]:
import numpy as np
import scipy.stats as stats

**2. Compute the Chi-Square Statistic:**

first ,compute the expected frequencies of all cells

Expected Frequencies (E): (ExpectedFrequency=(RowTotal*ColumnTotal)/N\)
EX: 120 * 240 / 600  = 48

Expected Frequencies means a type of probability count that can be calculated using probability theory

Repeat this for all cells.

After computing all expected frequencies, use the Chi-Square formula to find the test statistic:
χ² = Σ [(O - E)² / E]


In [ ]:
row1 = np.array([50,70])
row2 = np.array([80,100])
row3 = np.array([60,90])
row4 = np.array([30,50])
row5 = np.array([20,50])
data = np.array([row1, row2, row3, row4, row5])
data

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [ ]:
total=np.sum(data)
total

600

In [ ]:
sum_r1=np.sum(row1)
sum_r2=np.sum(row2)
sum_r3=np.sum(row3)
sum_r4=np.sum(row4)
sum_r5=np.sum(row5)
sum_row=np.array([sum_r1,sum_r2,sum_r3,sum_r4,sum_r5])
print(sum_row)

[120 180 150  80  70]


In [ ]:
sum_col =row1+row2+row3+row4+row5
sum_col

array([240, 360])

In [ ]:
exp =[]
for i in sum_row:
    for j in sum_col:
        value=(i*j)/total
        exp.append(value)

In [ ]:
exp

[48.0, 72.0, 72.0, 108.0, 60.0, 90.0, 32.0, 48.0, 28.0, 42.0]

In [ ]:
obj=np.array([50,70,80,100,60,90,30,50,20,50])
obj

array([ 50,  70,  80, 100,  60,  90,  30,  50,  20,  50])

In [ ]:
len(exp)

10

In [ ]:
len(obj)

10

In [ ]:
chi_cal = np.sum(np.square(obj-exp)/exp)
chi_cal

5.638227513227513

**3. Determine the Critical Value:**

degree of freedom (df) = (Number of Rows - 1) * (Number of Columns - 1)

In this case, df = (5-1) * (2-1) = 4.

significance level (α) of 0.05 and 4 degrees of freedom.
critical value in a Chi-Square distribution table value is 9.488.

In [ ]:
chi_tab=stats.chi2.ppf(1-0.05,df=4)
chi_tab

9.487729036781154

**4. Make a Decision**

Compare the Chi-Square statistic(chi_calc) with the critical value(chi_table):

If 𝜒2 > Critical Value, reject the null hypothesis.

If 𝜒2 ≤ Critical Value, fail to reject the null hypothesis.

In [ ]:
if chi_cal < chi_tab: #5.638 < 9.487
    print("Accept H0")
    print("There is no assocation")
else:
  print("Accept H1")
  print("There is an assocation")

Accept H0
There is no assocation
